### ISSUE TITLE
Enhancement: create vimdoc for CONFIG.md

### ISSUE THREAD

**Issue by matu3ba** [neovim/nvim-lspconfig#1228]
You could use the same script as the [nvim-lua-guide](https://github.com/nanotee/nvim-lua-guide/blob/master/doc/nvim-lua-guide.txt) for this.

**Comment by justinmk** [comment]
CONFIG.md is already generated, so whatever generates it can also generate the vimdoc.

**Comment by mjlbach** [comment]
I wrote a markdown -> vimscript treesitter based conversion pipeline which is used to generate our current `:help` doc, it could be extended to do this for config.md. Otherwise yeah, the docgen is going to need to get modified to output both markdown and vim doc.

**Comment by justinmk** [comment]
Since we don't really need (or want) help tags for the various things in CONFIG.md, all we really need is

1.  move CONFIG.md to `docs/CONFIG.md` (this makes sense semantically, too)
    - then `gf` works whereever CONFIG.md is mentioned in the vimdoc 
2.  add 1 new `:help` tag explicit in the current vimdoc, something like `lspconfig-index`:
    ```
    CONFIGURATIONS INDEX           *lspconfig-index*
    The CONFIG.md file contains the full list of included configurations.
    ```

**Comment by mjlbach** [comment]
I'm not sure it would be the worst thing in the world to generate a helptag for each language server.

**Comment by justinmk** [comment]
Let's at least prefix them with `lspconfig-*`. This is going to be a lot of things showing up in `:help` tab-completion.

**Comment by clason** [comment]
or `lsp-server-<foo>`?

**Comment by mjlbach** [comment]
This should be closed by #1384.

I don't think autogenerated the vimdoc for each server is a good idea, it's extremely difficult to navigate and would require setting a bunch of folds due to the extensive use of <details>
